corehq/apps/translations/tasks.py

```diff
@@ -137,3 +138,28 @@ def backup_project_from_transifex(domain, data, email):
         filename = "%s-TransifexBackup.zip" % project_details.get('name')
         email.attach(filename=filename, content=tmp.read())
         email.send()
+
+
+@task
+def download_project_from_hq(domain, data, email):
+    """Emails the requester with an excel file translations to be sent to Transifex.
+
+    Used to verify translations before sending to Transifex
+    """
+    lang = data.get('source_lang')
+    project_slug = data.get('transifex_project_slug')
+    gen = AppTranslationsGenerator(domain, data.get('app_id'), data.get('version'), lang, lang, 'default_')
+    parser = TranslationsParser(gen)
```

Comment:
`TranslationsParser` gets the translations [*from*](https://github.com/dimagi/commcare-hq/blob/6433924729335e1552481c1138dd23391aeed9d8/corehq/apps/translations/integrations/transifex/parser.py#L154) transifex. So this won't be using the gen you are passing.

You don't need parser. `gen` would give you all the rows in `_translation_data` and you have headers on `gen` as a dict or it might be the same as `expected_bulk_app_sheet_headers`. Then you can just generate an excel file using that similar to what we do in `download_bulk_app_translations`.

Reply:
Guys, every time you spell "Excel" with a lower-case "e", [Microsoft Cortana][1] puts another black mark against your name on its watchlist of subversives.

  [1]: https://www.microsoft.com/en-us/cortana